[Node Watch] Fix node watch error windows

Manvi-Agrawal opened the pull request:
### Issue
```
Error: spawn EINVAL
    at ChildProcess.spawn (node:internal/child_process:421:11)
    at spawn (node:child_process:761:9)
    at runWatcher (file:///C:/Users/manvi/Documents/qsharp/watch.mjs:108:22)
    at file:///C:/Users/manvi/Documents/qsharp/watch.mjs:121:1
    at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:323:24)
    at async loadESM (node:internal/process/esm_loader:28:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12) {
  errno: -4071,
  code: 'EINVAL',
  syscall: 'spawn'
}
```
### Fix
As per https://github.com/nodejs/node/issues/52554#issuecomment-2060026269, we should set `{shell: true}` because
spawning .cmd files is blocked.

`const npmCmd = isWin ? "npm.cmd" : "npm";`

### Verification
- Watch is working on Windows after my change.
- Tested on codespaces(Linux environment), and observed no side effect.

### Possible Side Effects
Need to investigate if `shell:True` can cause issues on Linux.

Manvi-Agrawal commented:
@tcNickolas , @minestarks , I was trying to use watch command on my windows machine(https://github.com/microsoft/qsharp/pull/1402#discussion_r1571199619) and realised that I couldnt use it. So, sending a small fix :-)

billti commented:
Wow. I tested this when I checked it in and was working fine, even did some special handling for Windows already. Reading through the linked Node.js issue, crazy that they broke this in a minor patch release 2 weeks ago and then just locked the conversation :-/

We'll need to test on macOS and Linux that the change works fine. Did you try it on any other OS?

Manvi-Agrawal commented on watch.mjs:
```diff
@@ -55,7 +55,7 @@ function buildRust() {
     "wasm-pack",
     ["build", "--dev", "--no-pack", "--target", "web", "--out-dir", buildDir],
-    { cwd: wasmDir },
+    { cwd: wasmDir, shell: true },
```
Need to investigate if `shell: true` can cause issues in Linux

Manvi-Agrawal commented:
> Wow. I tested this when I checked it in and was working fine, even did some special handling for Windows already. Reading through the linked Node.js issue, crazy that they broke this in a minor patch release 2 weeks ago and then just locked the conversation :-/

> We'll need to test on macOS and Linux that the change works fine. Did you try it on any other OS?

@billti I have only Windows :-). Maybe I can use codespaces/WSL for linux testing. I realised after raising PR, and updated description main comment and added a new comment to realise that you have this doubt :)

minestarks reviewed (approved):
LGTM but I haven't validated on macOS
